server/src/main/java/io/crate/analyze/PartitionPropertiesAnalyzer.java

```diff
@@ -60,25 +60,51 @@ public static PartitionName toPartitionName(RelationName relationName, List<Assi
         return new PartitionName(relationName, List.of(values));
     }
 
+
+    /**
+     * Return a {@link PartitionName} for the given partition properties.
+     * Returns null if the partitionProperties are empty.
+     *
+     * @throws PartitionUnknownException if the partition is missing from the table
+     * @throws IllegalArgumentException if the table is not partitioned, or if the properties don't match the partitionBy clause
+     */
+    @Nullable
     public static PartitionName toPartitionName(DocTableInfo tableInfo,
-                                                List<Assignment<Object>> partitionProperties) {
+                                                List<Assignment<Object>> partitionsProperties) {
+        PartitionName partitionName = toPartitionNameUnsafe(tableInfo, partitionsProperties);
+        if (partitionName == null) {
+            return null;
+        }
+        if (tableInfo.partitions().contains(partitionName) == false) {
+            throw new PartitionUnknownException(partitionName);
+        }
+        return partitionName;
+    }
+
+    /**
+     * Like {@link #toPartitionName(DocTableInfo, List) but doesn't raise a
+     * PartitionUnknownException if the table doesn't contain the partition
+     */
+    @Nullable
+    public static PartitionName toPartitionNameUnsafe(DocTableInfo tableInfo, List<Assignment<Object>> partitionsProperties) {
+        if (partitionsProperties.isEmpty()) {
```

Comment:
Shouldn't this check be moved after the next 2 that throw exceptions?

Reply:
No it's intentionally above, if there are no partitionProperties it can be a regular table, which is fine.
If we'd move it down, we'd have to do both checks on the call-sites:  check partition properties upfront if they're empty, and do a null check on the result. This way they only have to check the result for null.

Reply:
On second thought - an alternative would be to change it into an error case if the partition properties are empty.
Then the call-sites have to check that, but do not need to bother with null because the result is no longer nullable

Reply:
I have a slight preference to handle as error, because the name is `toPartitionName`,but not insisting.

Reply:
Pushed a fixup and removed the check, it's implicitly checked by `if (partitionsProperties.size() != tableInfo.partitionedBy().size())`

Reply:
thx!